lib/runnable-hostname.js

```diff
@@ -3,22 +3,33 @@
 var exists = require('101/exists');
 var createHostLabel = require('./create-host-label.js');
 var replaceInvalid = require('./replace-invalid.js');
+var regexForIsolationPrefix = /^[A-z0-9]*--/;
 
 module.exports = {
   elastic: elastic,
   direct:  direct
 };
 
+/**
+ * Returns true if the instance is isolated, but is not the isolation master
+ * @param opts
+ */
+function isIsolatedContainer (opts) {
+  return opts.isolated && !opts.isIsolationGroupMaster
+}
+
 function elastic (opts) {
   validateOpts(opts);
   // consistency btw direct and elastic
-  return direct(opts).replace(opts.shortHash+'-', '').toLowerCase();
+  // If the instance is isolated,
+  var replacement = isIsolatedContainer(opts) ? regexForIsolationPrefix : opts.shortHash + '-';
+  return direct(opts).replace(replacement, '').toLowerCase();
 }
 
 function direct (opts) {
   validateOpts(opts);
   var template = '{shortHash}-{repoName}-{env}-{ownerUsername}'
-    .replace('{shortHash}', opts.shortHash)
+    .replace('{shortHash}-', isIsolatedContainer(opts) ? '' : opts.shortHash + '-')
```

Comment:
before this was just ` opts.shortHash` but now you are making it null if this is an isolated container?